GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -0,0 +1,169 @@
+                                                                       error:nil];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNotNil(authFlow.authState);
+  XCTAssertNil(authFlow.error);
+}
+
+- (void)testGenerateAuthFlowFromAuthorizationResponse_noCode {
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  NSDictionary<NSString *, NSString *> *errorDict =
+      @{ NSLocalizedDescriptionKey : @"Unknown error" };
+  NSError *expectedError = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                               code:GIDVerifyErrorCodeUnknown
+                                           userInfo:errorDict];
+
+  OIDAuthorizationResponse *authorizationResponse = 
+      [OIDAuthorizationResponse testInstanceNoAuthCodeWithAdditionalParameters:nil 
+                                                                   errorString:nil];
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:authorizationResponse
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:nil];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqual(authFlow.error.code, expectedError.code);
+  XCTAssertEqualObjects(authFlow.error.userInfo[NSLocalizedDescriptionKey],
+                        expectedError.userInfo[NSLocalizedDescriptionKey]);
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+}
+
+- (void)testGenerateAuthFlowWithMissingAuthorizationResponse {
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  NSError *error = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                       code:GIDVerifyErrorCodeUnknown
+                                   userInfo:nil];
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:error];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqual(authFlow.error.code, error.code);
```

Comment:
I think maybe it would be better to assert that the two errors are the same.

Reply:
Done.